main.cpp

```diff
@@ -56,50 +56,64 @@ Purpose:  This project will show you the difference between member functions and
 
 #include <iostream>
 #include <string>
+#include <cmath>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    int value;
+    const char* name;
+    T(int v, const char* objectName) :   //1
+    value(v), //2
+    name(objectName)//3
+    {
+        std::cout << "Created object of class T with value " << value << " and name " << name << "." << std::endl;
+    }
 };
 
-struct <#structName1#>                                //4
+struct Comparator                                //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b) // 5
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
+        if (a != nullptr && b != nullptr)
+        {
+            if( a->value < b->value ) return a;
+            if( a->value > b->value ) return b;
+        }
         return nullptr;
     }
 };
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
-    {
-        
-    }
+    float target { 0 }, currentValue { 0 };
+    float updateAndMultiply(float* updatedValuePtr);      //12
 };
 
-struct <#structname2#>
+struct Updater
 {
-    static <#returntype#> <#staticFunctionA#>(U* that, <#type name#>* <#updatedValue#> )        //10
+    static float updateAndMultiply(U* that, float* updatedValuePtr ) //10
     {
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        if (that == nullptr || updatedValuePtr == nullptr) return std::nanf("");
+        
+        std::cout << "U's currentValue value: " << that->target << std::endl;
+        that->target = *updatedValuePtr;
+        std::cout << "U's target updated value: " << that->target << std::endl;
+        
+        while( std::abs(that->currentValue - that->target) > 0.001f )
         {
             /*
-             write something that makes the distance between that-><#name2#> and that-><#name1#> get smaller
+             write something that makes the distance between that->currentValue and that->target get smaller
              */
-            that-><#name2#> += ;
+            that->currentValue += (that->target - that->currentValue) * 0.5f;
         }
-        std::cout << "U's <#name2#> updated value: " << that-><#name2#> << std::endl;
-        return that-><#name2#> * that-><#name1#>;
+        std::cout << "U's currentValue updated value: " << that->currentValue << std::endl;
+        return that->currentValue * that->target;
     }
 };
+
+float U::updateAndMultiply(float* updatedValuePtr)
+{
+    return Updater::updateAndMultiply(this, updatedValuePtr);
```

Comment:
LOL this is really clever.  I've never seen anyone think of this solution.
I'll allow it.
- [ ] nice work

Reply:
Keeping things DRY ;) and admittedly being a bit lazy!